### PR TITLE
Closes #251

### LINKED ISSUE #251: Create dataset loader for SciQ
## Adding a Dataset
- **Name:** SciQ
- **Description:** *None provided*
- **Task:** QA
- **Paper:** https://aclanthology.org/W17-4413/
- **Data:** https://allenai.org/data/sciq
- **License:** ?

### PR DESCRIPTION
**Pull request by j-chim**
Please name your PR after the issue it closes. You can use the following line: "Closes #ISSUE-NUMBER" where you replace the ISSUE-NUMBER with the one corresponding to your dataset.

If the following information is NOT present in the issue, please populate:

- **Name:** *name of the dataset*
- **Description:** *short description of the dataset (or link to social media or blog post)*
- **Paper:** *link to the dataset paper if available*
- **Data:** *link to the online home of the dataset*

### Checkbox

- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `biodatasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_BIGBIO_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `BigBioConfig` for the source schema and one for a bigbio schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_bigbio biodatasets/my_dataset/my_dataset.py`.

### CODE REVIEW

**Comment by j-chim** [comment]
This dataset contains multiple-choice questions, some with contexts. 

1. Am unclear about question `type'. I've left it as an empty string for the moment since it's not provided in the source dataset. The value should be something like "multiple-choice", but in my opinion it could be confusing to identify them as MC given the lack of support for this question format at this time. 
2. To fit into the current bigbio QA schema, I've only selected `correct_answer` and have discarded all distractors. Are there better alternatives?

File: biodatasets/sciq/sciq.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+            document_id = split
+            for i, d in enumerate(data):
+                id = f"{split}_{i}"
+                yield id, {
+                    "id": id,
+                    "question_id": id,
+                    "document_id": document_id,
+                    "question": d["question"],
+                    "type": "",  # TODO - not provided in original data
+                    "context": d["support"],
```
Comment (galtay) [inline biodatasets/sciq/sciq.py:152]:
```suggestion
                    "context": d["support"],
                    "choices": "<List[str] of choices>"
```
@j-chim can you fill in the choices here from the datase`d`?

**Review by galtay — CHANGES_REQUESTED** [review]
Hi @j-chim , if you pull the latest version of `schemas.qa_features` and `tests/test_bigbio.py`, you should be able to use the Q/A type by, 
* adding a `choices` attribute to the bigbio output
* setting `type` to "multiple_choice"

File: biodatasets/sciq/sciq.py (hunk 1 above)
Comment (j-chim) [reply biodatasets/sciq/sciq.py:152]: thanks @galtay! both should be done in the latest commit

**Review by galtay — APPROVED** [review]
Looks good, thanks!

**Pull request merged**